### PR TITLE
feat: [CDS-47425]: added new env component / integrations of svc and env component with studio and RPF's

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- New Environment Component Added for scoped references
- Environment Component Integration with Studio and Runtime forms
- Service component integration added  in Run pipeline forms

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx [hunk 1]
```diff
@@ -357,36 +361,64 @@ export default function DeployEnvironment({
...
-              }
-            }}
-            multiTypeProps={{
-              onTypeChange: setEnvironmentsType,
-              width: 280,
-              allowableTypes: getAllowableTypesWithoutExpression(allowableTypes)
-            }}
-          />
-        ) : (
+          GLOBAL_SERVICE_ENV && !isUnderEnvGroup ? (
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx:364]: add comment here

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx [hunk 2]
```diff
@@ -378,50 +382,83 @@ export default function DeployEnvironmentEntityInputStep({
...
-            onChange={handleEnvironmentsChange}
-            multiTypeProps={{
-              width: 300,
-              height: 32,
-              allowableTypes
-            }}
-          />
+          GLOBAL_SERVICE_ENV ? (
+            <MultiTypeEnvironmentField
+              tooltipProps={{ dataTooltipId: 'specifyYourEnvironments' }}
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx:429]: would it be better to use commonProps object and extend it here? commonProps

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironment/DeployEnvironment.tsx:364]: Done

File: src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/DeployEnvironmentEntityStep/DeployEnvironmentEntityInputStep.tsx:429]: Done

**Review by AshwinBhatkal-Harness — DISMISSED** [review]

**Review by kapilkumawat86 — DISMISSED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Pull request merged**